source/MaterialXGenMdl/mdl/materialx/sampling.mdl

```diff
@@ -49,11 +51,13 @@ export float2 mx_compute_sample_size_uv(float2 uv, float filterSize, float filte
 //
 export float3 mx_normal_from_samples_sobel(float[9] S, float scale)
 {
-   float nx = S[0] - S[2] + (2.0*S[3]) - (2.0*S[5]) + S[6] - S[8];
-   float ny = S[0] + (2.0*S[1]) + S[2] - S[6] - (2.0*S[7]) - S[8];
-   float nz = scale * ::math::sqrt(1.0 - nx*nx - ny*ny);
-   float3 norm = ::math::normalize(float3(nx, ny, nz));
-   return (norm + 1.0) * 0.5;
+    // this produces NaN
+    float nx = S[0] - S[2] + (2.0*S[3]) - (2.0*S[5]) + S[6] - S[8];
+    float ny = S[0] + (2.0*S[1]) + S[2] - S[6] - (2.0*S[7]) - S[8];
+    // float nz = scale * ::math::sqrt(1.0 - nx*nx - ny*ny);
```

Comment:
This code is ends up being part of the data library - right? if so then it will be installed in locations that aren't attached to any git repo - so perhaps this is a case where keeping comments around that help keep the code more readable is a reasonable step?   

git history is not always the easiest place to find documentation.